src/charm.py

```diff
@@ -509,7 +527,9 @@ def _on_config_changed(self, _event: ConfigChangedEvent) -> None:
             self._stored.token = None
 
         if self.config["path"] != self._stored.path:
-            prev_config_for_flush["path"] = str(self._stored.path)
+            prev_config_for_flush["path"] = parse_github_path(
+                self._stored.path, self.config["group"]
```

Comment:
This would be need be changed when we remove `StoredSet`.
The `group` is only needed due to `self._stored.path` is a string. The `parse_github_path` is not needed when we use `CharmState` for persistence as it would directly provide a `GithubPath` object.
This is temp code until we remove `StoredSet`.